templates/prometheuskuberbacproxyconfig.go

```diff
@@ -0,0 +1,52 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+var KubeRBACProxyConfig = struct {
+	Authorization struct {
+		Static []struct {
+			Path            string `yaml:"path"`
+			ResourceRequest bool   `yaml:"resourceRequest"`
+			Verb            string `yaml:"verb"`
+		} `yaml:"static"`
+	} `yaml:"authorization"`
+}{
+	Authorization: struct {
+		Static []struct {
+			Path            string "yaml:\"path\""
+			ResourceRequest bool   "yaml:\"resourceRequest\""
+			Verb            string "yaml:\"verb\""
+		} "yaml:\"static\""
+	}{
+		Static: []struct {
+			Path            string "yaml:\"path\""
+			ResourceRequest bool   "yaml:\"resourceRequest\""
+			Verb            string "yaml:\"verb\""
+		}{
+			{
+				Path:            "/metrics",
+				ResourceRequest: false,
+				Verb:            "get",
+			},
+			{
+				Path:            "/federate",
+				ResourceRequest: false,
+				Verb:            "get",
+			},
+		},
+	},
```

Comment:
A couple of notes: 
1. Why are we creating our own struct and using one provided by the lib?
2. Can't we just have a template for the entire config map and not just the config itself?

Reply:
We can't use the struct provided by the library because of two reasons
1. ResourceRequest needs to be set to false but when we try to Marshal it ResourceRequest is removed as the default is set to false. Usually, they should have used pointer to the boolean variable but they haven't so if try to set the default value Marshal just removes it.  I'm unclear then please refer to this: https://stackoverflow.com/questions/37756236/json-golang-boolean-omitempty
2. The outermost key-value pair in the json should be authorization but it is not present in the struct they have provided so the container goes in failed state. So basically it is marshalled into `{
      "static":[
        {
          "verb":"get",
          "resourceRequest":false,
          "path":"/metrics"
        },
        {
          "verb":"get",
          "resourceRequest":false,
          "path":"/federate"
        }
      ]
    }`
    whereas it should marshal as `{
      "authorization":
      {
        "static":[
          {
            "verb":"get",
            "resourceRequest":false,
            "path":"/metrics"
          },
          {
            "verb":"get",
            "resourceRequest":false,
            "path":"/federate"
          }
        ]
      }
    }`
    
    I have used the entire configMap in the template file in my latest commit